channel-api/src/main/java/com/synopsys/integration/alert/channel/api/convert/BomComponentDetailConverter.java

```diff
@@ -48,27 +49,35 @@ public BomComponentDetailConverter(ChannelMessageFormatter formatter) {
     public List<String> gatherBomComponentPieces(BomComponentDetails bomComponent) {
         List<String> bomComponentSectionPieces = new LinkedList<>();
 
+        List<String> preConcernSectionPieces = gatherPreConcernSectionPieces(bomComponent);
```

Comment:
What is a preConcern?

Reply:
The concern section is related to "component concerns" (i.e. policy violations and vulnerabilities). The pre-concern section is the section before the component concerns.

Reply:
Is a pre-concern specific to the BomComponent conversion at this point?   Or is this something in other converters as well?  From looking at the code here in this class we're defining the component data which includes the component name and version, as well as, component attributes.  The variable name with preconcern had me asking the same question.  What's a preconcern?

If the variable name containing preconcern is confusing, then could we name it componentSectionPieces?  The componentSectionPieces contains the component header and the component attributes and the component concerns section is exactly that the component concerns.  The naming of variables is the hard part.  If renaming the variable would help avoid confusion.

Or if the intent isn't clear then could we split gatherPreConcernSectionPieces into two methods:
`List<String> gatherComponentHeaderPieces(AbstractBomComponentDetails bomComponent)`
That method executes the code existing in the 64 -73 to get the component name and version.
Then have another method:
`List<String> gatherComponentAttributePieces(AbstractBomComponentDetails bomComponent)`
That executes lines 75-79 to create all the component attribute data.

The last step would be to then change the method to execute this:
```
  List<String> componentHeaderPieces = gatherComponentHeaderPieces(bomComponent);
  List<String> componentAttributesPieces = gatherComponentAttributePieces(bomComponent);
  List<String> componentConcernSectionPieces = createComponentConcernSectionPieces(bomComponent);
  bomComponentSectionPieces.addAll(componentHeaderPieces);
  bomComponentSectionPieces.addAll(componentAttributesPieces);
  bomComponentSectionPieces.addAll(componentConcernSectionPieces);
```

Reply:
That's my suggestion if the preconcern variable is causing confusion.

Reply:
Yeah, when I wrote it, I was reading it as "the section before the concern-section" not "the section related to 'preconcerns'". I see that it is causing some confusion, so I'll try to make the variables a little more clear.